tests/test_pex_warnings.py

```diff
@@ -0,0 +1,76 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import warnings
+
+import pytest
+
+from pex import pex_warnings
+from pex.compatibility import PY2
+from pex.pex_info import PexInfo
+from pex.pex_warnings import PEXWarning
+from pex.variables import Variables
+
+
+def exercise_warnings(pex_info, **env):
+  with warnings.catch_warnings(record=True) as events:
+    warnings.resetwarnings()
+    pex_warnings.configure_warnings(pex_info, env=Variables(environ=env))
+    pex_warnings.warn('test')
+    return events
+
+
+def assert_warnings(pex_info, **env):
+  events = exercise_warnings(pex_info, **env)
+  assert 1 == len(events)
+  warning = events[0]
+  assert PEXWarning == warning.category
+  assert 'test' == str(warning.message)
+
+
+def assert_no_warnings(pex_info, **env):
+  events = exercise_warnings(pex_info, **env)
+  assert 0 == len(events)
+
+
+def pex_info_no_emit_warnings():
+  pex_info = PexInfo.default()
+  pex_info.emit_warnings = False
+  return pex_info
+
+
+skip_py2_warnings = pytest.mark.skipif(PY2,
+                                       reason="The warnings.catch_warnings mechanism doesn't work "
+                                              "properly under python 2.7 / pypy2, etc. across "
+                                              "multiple tests")
+
```

Comment:
And helpful to reiterate this is not an issue with our code and things work IRL, only this is an issue with the testing mechanism, as you explained to me over Slack.

Reply:
I mean, I already say this.

If you have some exact wording changes you want me to toss in, I'll do it.

Reply:
Maybe something like: 
```
"The warnings.catch_warnings mechanism doesn't work "
"properly under python 2.7 / pypy2, etc. across "
"multiple tests. This is an issue with the testing mechanism due to"
"issues with the underlying C code, and it does not negatively impact"
"our warning code when used outside of these tests."
```

Make it clear no one should waste time on trying to fix this.

Reply:
OK - left out the C reference since that doesn't explain pypy and I still don't think the underlying reason is important. I did add language though that makes it clear this is only a unit-test failure that can't be fixed and is otherwise covered by ITs. I will point out we should never have to say - but this actually works - in comments. If we're allowing code to hit master that doesn't work - we have big problems. It should be assumed!